example/example.dart

```diff
@@ -24,11 +24,11 @@ num parseNumber(String source) {
 
   // [Scanner.lastMatch] holds the [MatchData] for the most recent call to
   // [Scanner.scan], [Scanner.expect], or [Scanner.matches].
-  var number = num.parse(scanner.lastMatch![0]);
+  var number = num.parse((scanner.lastMatch![0])!);
```

Comment:
Those parentheses shouldn't be necessary. If they are, it's a parser bug. Can you file an issue in that case? Likewise below.

Reply:
It looks like just an analyzer thing, the error I actually get without them is:

`[dart] The argument type 'dynamic' can't be assigned to the parameter type 'String'.`

I can create a repro and file something.